R/scATAC.Peak.annotation.R

```diff
@@ -641,21 +703,32 @@ peaks_closest_gene <- function(peaks, annotations=NULL, gene_element=NULL, TSSmo
   rownames(peaks_close) <- peaks_nam_not_annotated
   cat(dim(peaks_close), "\n")
   if (fast == F) {
-  if (TSSmode) {
-    cat(dim(peaks_annotated_TSS_ol), "\n")
-    cat(dim(peaks_annotated_TSS_on), "\n")
-    peaks_on_gene <- rbind(peaks_annotated_TSS_ol, peaks_annotated_TSS_on)
-    peaks_on_gene <- rbind(peaks_on_gene, peaks_annotated_end)
-    cat(dim(peaks_on_gene), "\n")
-    peaks_close <- rbind(peaks_on_gene, peaks_close)
-    cat(dim(peaks_close), "\n")
-  }
-  else{
-    cat(dim(peaks_annotated), "\n")
-    cat(dim(peaks_close), "\n")
-    peaks_close <- rbind(peaks_annotated, peaks_close)}
+    if (TSSmode) {
+      cat(dim(peaks_annotated_TSS_ol), "\n")
+      cat(dim(peaks_annotated_TSS_on), "\n")
+      peaks_on_gene <- rbind(peaks_annotated_TSS_ol, peaks_annotated_TSS_on)
+      peaks_on_gene <- rbind(peaks_on_gene, peaks_annotated_end)
+      cat(dim(peaks_on_gene), "\n")
+      peaks_close <- rbind(peaks_on_gene, peaks_close)
+      cat(dim(peaks_close), "\n")
+    }
+    else{
```

Comment:
this else has to be removed. As the colnames should be the same in fast mode or not. Fast mode just throws out the peaks which have a match after peak_on_genes